Sources/SourceKitBazelBSP/Server/MessageHandler/BSPMessageHandler.swift

```diff
@@ -35,10 +35,6 @@ final class BSPMessageHandler: MessageHandler {
         var notificationHandlers: [String: AnyNotificationHandler] = [:]
     }
 
-    // We currently use a single-threaded setup for simplicity,
-    // but we can eventually reply asynchronously if we find a need for it.
-    private let lock: OSAllocatedUnfairLock<Void> = .init()
-    // FIXME: Can't put state into the lock for now because of recursiveness in the registration.
     nonisolated(unsafe) private var state: State = State()
```

Comment:
Locking should be unnecessary due to thread safety on sourcekit-lsp's side: https://github.com/swiftlang/sourcekit-lsp/blob/d18aba71ce3c539ec1474470228ab7fea9b66105/Sources/LanguageServerProtocolJSONRPC/JSONRPCConnection.swift#L55-L59

Will only be necessary on our end if we start running things asynchronously.